src/lib/abacus/dydxChainTransactions.ts

```diff
@@ -229,6 +233,77 @@ class DydxChainTransactions implements AbacusDYDXChainTransactionsProtocol {
     }
   }
 
+  async simulateWithdrawTransaction(params: HumanReadableWithdrawPayload): Promise<string> {
+    if (!this.compositeClient || !this.localWallet) {
+      throw new Error('Missing compositeClient or localWallet');
+    }
+
+    const { subaccountNumber, amount } = params ?? {};
+    const compositeClient = this.compositeClient;
+    const subaccountClient = new SubaccountClient(this.localWallet, subaccountNumber);
+
+    try {
+      const tx = await compositeClient.simulate(
+        this.localWallet,
+        () =>
+          new Promise((resolve) => {
+            const msg = compositeClient.withdrawFromSubaccountMessage(subaccountClient, amount);
+
+            resolve([msg]);
+          }),
+        undefined
+      );
+
+      const parsedTx = this.parseToPrimitives(tx);
+
+      return JSON.stringify(parsedTx);
+    } catch (error) {
+      log('DydxChainTransactions/simulateWithdrawTransaction', error);
+
+      return JSON.stringify({
+        error,
+      });
+    }
+  }
+
+  async simulateTransferNativeTokenTransaction(params: HumanReadableTransferPayload): Promise<string> {
+    if (!this.compositeClient || !this.localWallet) {
+      throw new Error('Missing compositeClient or localWallet');
+    }
+
+    const { subaccountNumber, amount, recipient } = params ?? {};
+    const compositeClient = this.compositeClient;
+    const subaccountClient = new SubaccountClient(this.localWallet, subaccountNumber);
+
+    try {
+      const tx = await compositeClient.simulate(
+        this.localWallet,
+        () =>
+          new Promise((resolve) => {
+            const msg = compositeClient?.validatorClient.post.composer.composeMsgSendToken(
+              subaccountClient.address,
+              recipient,
+              DYDX_DENOM,
+              Long.fromNumber(amount * QUANTUM_MULTIPLIER)
+            );
+
+            resolve([msg]);
+          }),
```

Comment:
i think it's missing the `GAS_PRICE_DYDX_DENOM`? (making sure it's taking gas in native tokens)

Reply:
good call!